src/biotite/structure/bonds.pyx

```diff
@@ -1645,12 +1647,37 @@ def connect_via_residue_names(atoms, bint inter_residue=True,
                 res_names[curr_start_i], {}
             )
 
+        # Check if we should use alternative atom names
         atom_names_in_res = atom_names[curr_start_i : next_start_i]
+        std_atom_ids = get_from_ccd(
+            "chem_comp_atom", 
+            res_names[curr_start_i], 
+            "atom_id"
+        ) 
+
+        # Only lookup alternative atom names if we cannot match
+        # all atoms using standard names
+        if (atom_names_in_res is not None and \
+            std_atom_ids is not None and \
+            not set(atom_names_in_res).issubset(std_atom_ids)):
```

Comment:
This line is executed for each residue, so performance is important here. Let's check if something like `np.isin(atom_names_in_res, std_atom_ids).all()` is faster, as soon as the benchmarks in the CI have run (tests need to pass first).